buffer.go

```diff
@@ -91,6 +92,11 @@ func (r *recordBuffer) Add(m *router.Message) error {
 		return err
 	}
 
+	// We default to a uuid if the template didn't match.
+	if pKey == "" {
+		pKey = uuid.New()
```

Comment:
I wouldn't do this if the template doesn't match - maybe if the template doesn't exist, but this could lead to mistakes if someone has a typo in their template.